packages/roosterjs-content-model/lib/publicApi/block/toggleMargins.ts

```diff
@@ -0,0 +1,29 @@
+import { getObjectKeys } from 'roosterjs-editor-dom';
+import { createParagraphDecorator } from '../../modelApi/creators/createParagraphDecorator';
+import { MarginFormat } from '../../publicTypes/format/formatParts/MarginFormat';
+import { IContentModelEditor } from '../../publicTypes/IContentModelEditor';
+import { formatParagraphWithContentModel } from '../utils/formatParagraphWithContentModel';
+
+/**
+ * Toggles the current block(s) margin properties.
+ * @param editor The editor to operate on
+ * @param marginFormat Object containing margin props to be changed.
+ * If a margin property is not in this object, it won't be changed.
+ */
+export default function toggleMargins(editor: IContentModelEditor, marginFormat: MarginFormat) {
+    formatParagraphWithContentModel(editor, 'toggleMargins', para => {
+        if (!para.decorator || para.decorator.tagName !== 'p') {
+            para.decorator = createParagraphDecorator('p');
```

Comment:
How do you want to handle header? Because tagName can also be h1-h6. What is the expected behavior when toggle margin on a header?